src/System.Sorting/tests/SortTests.cs

```diff
@@ -0,0 +1,140 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Reflection;
+using System.Linq;
+using Xunit;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace System.Sorting.Tests
+{
+    public class SortTests
+    {
+        [Fact]
+        public static void TestSort()
+        {
+            IComparer<int> icomparer = new IntegerComparer();
+
+            TestSortHelper<int>(new int[] { }, 0, 0, icomparer);
+            TestSortHelper<int>(new int[] { 5 }, 0, 1, icomparer);
+            TestSortHelper<int>(new int[] { 5, 2 }, 0, 2, icomparer);
+
+            TestSortHelper<int>(new int[] { 5, 2, 9, 8, 4, 3, 2, 4, 6 }, 0, 9, icomparer);
+            TestSortHelper<int>(new int[] { 5, 2, 9, 8, 4, 3, 2, 4, 6 }, 3, 4, icomparer);
+            TestSortHelper<int>(new int[] { 5, 2, 9, 8, 4, 3, 2, 4, 6 }, 3, 6, icomparer);
+
+            IComparer<string> scomparer = new StringComparer();
+            TestSortHelper<String>(new string[] { }, 0, 0, scomparer);
+            TestSortHelper<String>(new string[] { "5" }, 0, 1, scomparer);
+            TestSortHelper<String>(new string[] { "5", "2" }, 0, 2, scomparer);
+
+            TestSortHelper<String>(new string[] { "5", "2", null, "8", "4", "3", "2", "4", "6" }, 0, 9, scomparer);
+            TestSortHelper<String>(new string[] { "5", "2", null, "8", "4", "3", "2", "4", "6" }, 3, 4, scomparer);
+            TestSortHelper<String>(new string[] { "5", "2", null, "8", "4", "3", "2", "4", "6" }, 3, 6, scomparer);
+        }
+
+
+        private static void TestSortHelper<T>(T[] array, int index, int length, IComparer<T> comparer)
+        {
+
+
+            T[] control = SimpleSort<T>(array, index, length, comparer);
+
+            //TODO - Can use Reflection to get the derivedTypes of Sorting and invoking the methods. 
```

Comment:
I call out later why I think this is a bad idea here but if you needed to do this at some other point you would want to use the `MakeGenericType` method on TypeInfo to substitute concrete values for the type parameters and then use `Activator.CreateInstance` to build an instance of that type.